src/main/scala/com/gradle/internal/CustomBuildScanEnhancements.scala

```diff
@@ -1,328 +1,369 @@
 package com.gradle.internal
 
 import com.gradle.develocity.agent.sbt.api.configuration.BuildScan
-import com.gradle.internal.CiUtils.*
-import com.gradle.internal.Utils.*
-import sbt.URL
+import com.gradle.develocity.agent.sbt.api.configuration.Server
+import com.gradle.internal.Utils.Env
+import java.util.Properties
+import java.net.URL
+import sbt.Logger
 
 /**
  * Adds a standard set of useful tags, links and custom values to all build scans published.
  */
-class CustomBuildScanEnhancements(serverConfig: ServerConfigTemp, scalaVersions: String) {
-
-  private val buildScan = new BuildScanTemp()
-
-  private val SYSTEM_PROP_IDEA_VENDOR_NAME = "idea.vendor.name"
-  private val SYSTEM_PROP_IDEA_VERSION = "idea.version"
-  private val SYSTEM_PROP_IDEA_MANAGED = "idea.managed"
-  private val SYSTEM_PROP_ECLIPSE_BUILD_ID = "eclipse.buildId"
-  private val SYSTEM_PROP_IDEA_SYNC_ACTIVE = "idea.sync.active"
+class CustomBuildScanEnhancements(serverConfig: Server, scalaVersions: Seq[String], logger: Logger)(implicit
```

Comment:
for my understanding, what is the difference between `logger` and `println` in `sbt`?

Reply:
`println` goes to standard output, whereas what happens to messages passed to the logger depends on the logger's configuration (eg. it could be hidden, be written somewhere else, log output can be later retrieved, etc.)